services/tone_analyzer/v3.js

```diff
@@ -0,0 +1,141 @@
+  var cfenv = require('cfenv');
+  var toneutils = require('../../utilities/tone-utils');
+
+  // Require the Cloud Foundry Module to pull credentials from bound service 
+  // If they are found then they are stored in sUsername and sPassword, as the 
+  // service credentials. This separation from sUsername and username to allow 
+  // the end user to modify the node credentials when the service is not bound.
+  // Otherwise, once set username would never get reset, resulting in a frustrated
+  // user who, when he errenously enters bad credentials, can't figure out why
+  // the edited ones are not being taken.
+
+  // Not ever used, and codeacy complains about it.
+
+  var username, password, sUsername, sPassword;
+
+  var service = cfenv.getAppEnv().getServiceCreds(/tone analyzer/i)
+
+  if (service) {
+    sUsername = service.username;
+    sPassword = service.password;
+  }
+
+
+  // Node RED Admin - fetch and set vcap services
+  RED.httpAdmin.get('/watson-tone-analyzer/vcap', function (req, res) {
+    res.json(service ? {bound_service: true} : null);
+  });
+
+
+  // Check that the credentials have been provided
+  // Credentials are needed for each the service.
+  var checkCreds = function(credentials) {
+    var taSettings = null;
+
+    username = sUsername || credentials.username;
+    password = sPassword || credentials.password;      
+
+    if (username && password) {
+      taSettings = {};
+      taSettings.username = username;
+      taSettings.password = password;
+    }
+
+    return taSettings;
+  }
+
+
+  // Function that checks the configuration to make sure that credentials,
+  // payload and options have been provied in the correct format.
+  var checkConfiguration = function(msg, node, cb) {
+    var message = null;      
+    var taSettings = null;
+
+    taSettings = checkCreds(node.credentials);
+
+    if (!taSettings) {
+      message = 'Missing Tone Analyzer service credentials';
+    } else if (msg.payload) {
+      message = toneutils.checkPayload(msg.payload);
+    } else  {
+      message = 'Missing property: msg.payload';
+    }
+
+    if (cb) {
+      cb(message, taSettings);
+    }
+  };
+
+
+  // function when the node recieves input inside a flow. 
+  // Configuration is first checked before the service is invoked.
+  var processOnInput = function(msg, config, node) {
+    checkConfiguration (msg, node, function(err, settings){
+      if (err) {
+        node.status({fill:'red', shape:'dot', text:err}); 
+        node.error(err, msg);
+        return;
+      } else {
+        var tone_analyzer = watson.tone_analyzer({
+          'username': settings.username,
+          'password': settings.password,
+          'version': 'v3',
+          'version_date': '2016-05-19'
+        });
+
+        var options = toneutils.parseOptions(msg, config);
+    
+        node.status({fill:'blue', shape:'dot', text:'requesting'});
+        tone_analyzer.tone(options, function (err, response) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'tone_analyzer' is not in camel case.](https://www.codacy.com/app/soheel_chughtai/chughts-node-red-node-watson/file/2690702778/issues/source?bid=3330826&fileBranchId=3339101#l107)